sdk/identity/azure-identity/inc/azure/identity/detail/client_credential_core.hpp

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "azure/identity/dll_import_export.hpp"
+
+#include <azure/core/credentials/credentials.hpp>
+#include <azure/core/url.hpp>
+
+#include <string>
+
+namespace Azure { namespace Identity { namespace _detail {
+  class ClientCredentialCore final {
```

Comment:
Why do you think it's valuable to have this private type rather than the fields be part of `ClientCredential` as they were previously?

What does "Core" imply here?

Reply:
This class does common parts of the job for both `ClientSecretCredential` and `ClientCertificateCredential`. It's functionality will be expanded even more with Multi-tenant/Challenge-based auth support. Normally it probably would've been an abstract base class havinf some methods implemented, but we can achieve the same by aggregating it, and using the functionality without inheritance (it does not override any virtual functions anyway). Plus, it keeps inheritance structure, that is visible to the SDK users, clean.